ui/checkboxradio.js

```diff
@@ -0,0 +1,287 @@
+		if ( this.options.disabled === true ){
+			this._setOption( "disabled", true );
+		}
+
+		this._readType();
+
+		this._enhance();
+
+		this._on({
+			"change": "_toggleClasses",
+			"focus": function() {
+				this.label.addClass( "ui-state-focus ui-visual-focus" );
+			},
+			"blur": function() {
+				this.label.removeClass( "ui-state-focus ui-visual-focus" );
+			}
+		});
+	},
+
+	_readType: function() {
+		this.type = this.element[ 0 ].type;
+		if ( !/radio|checkbox/.test( this.type ) ) {
+			throw new Error( "Can't create checkboxradio widget for type " + this.type );
+		}
+	},
+
+	_readLabel: function() {
+		var ancestor, labelSelector, parent = this.element.closest( "label" );
+		// Check control.labels first
+		if ( this.element[ 0 ].labels !== undefined && this.element[ 0 ].labels.length > 0 ){
+			this.label = $( this.element[ 0 ].labels[ 0 ] );
+		} else if ( parent.length > 0 ) {
+			this.label = parent;
+		} else {
+
+			// We don't search against the document in case the element
+			// is disconnected from the DOM
+			ancestor = this.element.parents().last();
+
+			// Look for the label based on the id
+			labelSelector = "label[for='" + this.element.attr( "id" ) + "']";
+			this.label = ancestor.find( labelSelector );
+			if ( !this.label.length ) {
+
+				// The label was not found make sure ancestors exist if they do check their siblings
+				// if they dont check the elements siblings
+				ancestor = ancestor.length ? ancestor.siblings() : this.element.siblings();
+
+				// Check if any of the new set of ancestors is the label
+				this.label = ancestor.filter( labelSelector );
+				if ( !this.label.length ) {
+
+					// Still not found look inside the ancestors for the label
+					this.label = ancestor.find( labelSelector );
+				}
+			}
+		}
+	},
+
+	_enhance: function() {
+		var checked = this.element.is( ":checked" );
+
+		this._updateIcon( checked );
+		this.element.addClass( "ui-helper-hidden-accessible " +
+			this._classes( "ui-checkboxradio" ) );
+
+		this.label.addClass( baseClasses + " " + this._classes( "ui-" + this.type + "-label" ) );
+
+		if ( checked ) {
+			this.label.addClass( this._classes( "ui-" + this.type + "-checked" ) +
+				" ui-state-active" );
+		}
+		if ( this.options.label && this.options.label !== this.originalLabel ) {
+			this.label.html( this.icon ? this.icon : "" ).append( this.options.label );
+		} else if ( this.originalLabel ) {
+			this.options.label = this.originalLabel;
+		}
+	},
+
+	widget: function() {
+		return this.label;
+	},
+
+	_toggleClasses: function() {
+		var checked = this.element.is( ":checked" );
+		this.label.toggleClass( this._classes( "ui-" + this.type + "-checked" ) +
+			" ui-state-active", checked );
+		if ( this.options.icon && this.type === "checkbox" ) {
+			this.icon
+				.toggleClass( "ui-icon-check", checked )
+				.toggleClass( "ui-icon-blank", !checked );
+		}
+		if ( this.type === "radio" ) {
+			radioGroup( this.element[0] )
```

Comment:
I'd keep it as is. Ideally this would be something like `element.radioGroup()`, but it doesn't really make sense for us to create a proper plugin and have to document it and add a dependency. The reason it doesn't exclude the current element is because it does what you'd expect: finds all radios associated with the current one. This method already existed and its scope didn't change.